GitHub workflows exhaust worker disk space

The [workflow failures](https://github.com/uncefact/tests-untp/actions/runs/13697421876/job/38302941062) in our GitHub Actions are occurring due to disk space limitations. GitHub only guarantees 14 GB of free disk space on their [standard runners](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories), with actual availability varying between workers.

## Raw logs
```
2025-03-06T11:23:26.4867661Z #30 [app dependencies 2/2] RUN yarn install && yarn cache clean
2025-03-06T11:23:41.2914017Z #30 116.4 [5/5] Building fresh packages...
2025-03-06T11:23:57.7509248Z ##[warning]You are running out of disk space. The runner will stop working when the machine runs out of disk space. Free space left: 0 MB
2025-03-06T11:44:56.0239438Z ##[error]The operation was canceled.
```

## Affected workflows
- [build_test.yml](https://github.com/uncefact/tests-untp/blob/next/.github/workflows/build_test.yml)
- [release.yml](https://github.com/uncefact/tests-untp/blob/next/.github/workflows/release.yml)

## Potential Solutions

### 1. Upgrade GitHub-Hosted Runner
GitHub offers [enhanced runner specifications](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/about-larger-runners#machine-sizes-for-larger-runners) that would resolve this issue, but this feature is only available for Teams and Enterprise accounts.

### 2. Self-Host the GitHub Runner
We could set up our own runners on cloud infrastructure (AWS, GCP, etc.), giving us complete control over hardware specifications.

### 3. Free Up Disk Space ([not viable](https://github.com/uncefact/tests-untp/actions/runs/13753608106/job/38457653489))
~~Implement third-party actions to clear disk space and remove node_modules directories/Docker images after each step. This approach would be somewhat cumbersome as three separate steps require these assets (application build/unit tests, end-to-end tests, and documentation build). The end-to-end test step appears to be the primary trigger for disk space exhaustion.~~

### 4. Separate Steps into Different Workflows ([not viable](https://github.com/uncefact/tests-untp/actions/runs/13753608106/job/38457653489))
~~Split high disk-consuming steps into separate workflows, each using a fresh runner. Initial testing indicates this would work, but it introduces complexity for the release process. Currently, a release is blocked if end-to-end tests fail, but this dependency would be lost with separate workflows. I'm unaware of the ability to depend on the status of another workflow without including it within the same runner.~~

Awaiting @kshychko's guidance on the preferred approach.

--- a/.github/workflows/build_test.yml
+++ b/.github/workflows/build_test.yml
@@ -50,6 +50,12 @@ jobs:
             UNTP test suite, ./packages/untp-test-suite/coverage/coverage-summary.json
             VC test suite, ./packages/vc-test-suite/coverage/coverage-summary.json
             UNTP Playground, ./packages/untp-playground/coverage/coverage-summary.json
+      
+      #https://github.com/actions/runner-images/issues/2840#issuecomment-1284059930
+      - name: Free up space on ubuntu runner for E2E tests
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
 
       - name: Start E2E docker compose
         run: SEEDING=true docker compose -f docker-compose.e2e.yml up -d

--- a/docker-compose.e2e.yml
+++ b/docker-compose.e2e.yml
@@ -147,6 +147,8 @@ services:
         CONFIG_FILE: ./e2e/cypress/fixtures/app-config.json
     ports:
       - '3003:80'
+    depends_on:
+      - untp-playground
 
 volumes:
   vckit-data:

